exporter/signalfxexporter/translation/converter.go

```diff
@@ -94,287 +83,273 @@ func NewMetricsConverter(logger *zap.Logger, t *MetricTranslator) *MetricsConver
 // MetricDataToSignalFxV2 converts the passed in MetricsData to SFx datapoints,
 // returning those datapoints and the number of time series that had to be
 // dropped because of errors or warnings.
-func (c *MetricsConverter) MetricDataToSignalFxV2(mds []consumerdata.MetricsData, sfxDataPoints []*sfxpb.DataPoint) ([]*sfxpb.DataPoint, int) {
-	var numDroppedTimeSeries int
-	var droppedDPCount int
-	for _, md := range mds {
-		sfxDataPoints, droppedDPCount = c.metricDataToSfxDataPoints(md, sfxDataPoints)
-		numDroppedTimeSeries += droppedDPCount
-	}
-	sanitizeDataPointDimensions(sfxDataPoints)
-	return sfxDataPoints, numDroppedTimeSeries
-}
-
-func (c *MetricsConverter) metricDataToSfxDataPoints(md consumerdata.MetricsData, sfxDataPoints []*sfxpb.DataPoint) ([]*sfxpb.DataPoint, int) {
+func (c *MetricsConverter) MetricDataToSignalFxV2(rm pdata.ResourceMetrics) ([]*sfxpb.DataPoint, int) {
 	var numDroppedTimeSeries int
 
-	// Labels from Node and Resource.
-	// TODO: Options to add lib, service name, etc as dimensions?
-	//  Q.: what about resource type?
-	nodeAttribs := md.Node.GetAttributes()
-	resourceAttribs := md.Resource.GetLabels()
+	sfxDatapoints := make([]*sfxpb.DataPoint, 0)
 
-	extraDimensions := make([]*sfxpb.Dimension, 0, len(nodeAttribs)+len(resourceAttribs))
+	res := rm.Resource()
+	for j := 0; j < rm.InstrumentationLibraryMetrics().Len(); j++ {
+		ilm := rm.InstrumentationLibraryMetrics().At(j)
+		if ilm.IsNil() {
+			continue
+		}
+		for k := 0; k < ilm.Metrics().Len(); k++ {
+			m := ilm.Metrics().At(k)
+			if m.IsNil() {
+				continue
+			}
 
-	extraDimensions = appendResourceAttributesToDimensions(extraDimensions, resourceAttribs)
-	extraDimensions = appendAttributesToDimensions(extraDimensions, nodeAttribs)
+			dps, droppedDPCount := c.metricToSfxDataPoints(m, res)
+			numDroppedTimeSeries += droppedDPCount
 
-	for _, metric := range md.Metrics {
-		if metric == nil || metric.MetricDescriptor == nil {
-			c.logger.Warn("Received nil metrics data or nil descriptor for metrics")
-			numDroppedTimeSeries += len(metric.GetTimeseries())
-			continue
+			sfxDatapoints = append(sfxDatapoints, dps...)
```

Comment:
Consider to add a TODO to preallocate the sfxDatapoints (or just do it) with number of points.

Reply:
Done.